interfaces/dotnet/Cantera/src/SpeciesCollection.cs

```diff
@@ -126,8 +108,12 @@ IEnumerator IEnumerable.GetEnumerator() =>
     /// Determines the index of the given <see cref="Species" /> in this collection,
     /// returning -1 if the species is not found.
     /// </summary>
-    public int IndexOf(Species species) =>
-        IndexOf(species.Name);
+    public int IndexOf(Species species)
+    {
+        ArgumentNullException.ThrowIfNull(species);
+
+        return IndexOf(species.Name);
```

Comment:
In theory, this duplicates features available in C++, see `Phase::speciesIndex` (which is exposed in Clib). Either is fine as long as the indices are consistent. 

One thing I'd like to point out is that C++ does allow for adding of `Species` via `Phase::addSpecies`, which needs to be consistent. I don't see this here, but I may be overlooking something?